test/jdk/java/io/File/TempDirDoesNotExist.java

```diff
@@ -25,91 +25,173 @@
-            testMessageExist(0, ioWarningMsg, "-Djava.io.tmpdir=" + tempDir,
-                    "TempDirDoesNotExist", "nio");
-
-            // test with security manager
-            testMessageExist(0, ioWarningMsg, "-Djava.io.tmpdir=" + tempDir
-                            + " -Djava.security.manager",
-                    "TempDirDoesNotExist", "io");
-
-            testMessageExist(0, ioWarningMsg, "-Djava.io.tmpdir=" + tempDir
-                            + " -Djava.security.manager",
-                    "TempDirDoesNotExist", "nio");
-
-            // error message printed only once
-            testMessageCounter(0, "-Djava.io.tmpdir=" + tempDir,
-                    "TempDirDoesNotExist", "io", "nio");
-        }
+    private static String tempDir() {
+        String timeStamp = String.valueOf(System.currentTimeMillis());
+        return Path.of(USER_DIR, "non-existing-", timeStamp).toString();
+    }
+
+    public static Stream<Arguments> existingProvider() {
+        List<Arguments> list = new ArrayList<Arguments>();
+
+        // invalid custom java.io.tmpdir
+        Arguments args = Arguments.of(0, WARNING,
+                                      new String[] {
+                                          "-Djava.io.tmpdir=" +
+                                          tempDir(),
+                                          "TempDirDoesNotExist", "io"
+                                      });
+        list.add(args);
+        args = Arguments.of(0, WARNING,
+                            new String[] {
+                                "-Djava.io.tmpdir=" + tempDir(),
+                                "TempDirDoesNotExist", "nio"
+                            });
+        list.add(args);
+
+        // test with security manager
+        args = Arguments.of(0, WARNING,
+                            new String[] {
+                                "-Djava.io.tmpdir=" + tempDir()
+                                + " -Djava.security.manager",
+                                "TempDirDoesNotExist", "io"
+                            });
+        list.add(args);
+        args = Arguments.of(0, WARNING,
+                            new String[] {
+                                "-Djava.io.tmpdir=" + tempDir()
+                                + " -Djava.security.manager",
+                                "TempDirDoesNotExist", "nio"
+                            });
+        list.add(args);
+
+        return list.stream();
+    }
+
+    public static Stream<Arguments> nonexistentProvider() {
+        List<Arguments> list = new ArrayList<Arguments>();
+
+        // standard test with default setting for java.io.tmpdir
+        Arguments args = Arguments.of(0, WARNING,
+                                      new String[] {
+                                          "TempDirDoesNotExist",
+                                          "io"
+                                      });
+        list.add(args);
```

Comment:
You can use 
```
Stream.of( Arguments.of(..),
           Arguments.of(...),
           ...);
```
A bit more compact and without the explicit arglist.

Reply:
So changed in cab8fa50a848f3c796bc81adcfd214aef0d054d3.